### ISSUE TITLE
How to treat background training images when fine tuning from COCO

### ISSUE THREAD

**Issue by pinkerltm** [pierluigiferrari/ssd_keras#113]
In SSD300 model you add a further class to account Background. My Training dataset ([Belgian traffic sign detection dataset](http://btsd.ethz.ch/shareddata/)) there are also instances of "Background" or traffic signs which do not belong to a certain class (Labelled: 0).

I thought of just truncating them from the training and validation sets, as the Label 0 isn't allowed anyway. But if I wanted to put them in the learning process too, to which label do I have to map them or how is background in MS COCO labelled?

**Comment by pierluigiferrari** [comment]
The background class is not a property of any given dataset, it's a property of SSD. MS COCO, Pascal VOC, or any other datasets I know of don't label a background class. The background class is just a way for SSD to assign a label to the boxes that are not any of the positive classes. Every predicted box needs to belong to some class. Since SSD predicts a fixed (large) number of boxes regardless of the number of actual objects in an image, all the boxes that don't detect any of the objects in the dataset need to be assigned some label. This is what the background class is for.

Your dataset is the first dataset I've ever heard of that labels a background class. I don't fully understand what's part of the definition of this background class in your dataset, but it wouldn't make any sense to try to treat it as the background class in the sense that SSD uses this background class. If you would want to explicitly include whatever is labeled as "background" boxes in your dataset, then you would just add them as another regular class. E.g. if your dataset has 80 non-background classes and that one background class, just make the background class number 81.

That being said, I don't understand how bounding boxes for a background class make any sense. A bounding box should enclose an object, and all bounding boxes of any given class should enclose objects of the same type. How is the model supposed to learn to predict the background class if the bounding boxes for that background class contain all sorts of things? Also, how does it make sense to put a bounding box around a patch of background? I.e. I don't understand how it makes sense to have a labeled background class in the dataset to begin with. Certainly nothing good can come from trying to train a model on such a background class because that data will probably be highly contradictory within itself.

**Comment by pinkerltm** [comment]
Thanks for your Statement, that pretty much backs up what I was thinking about it, but I was still too unsure about due to my lack of experience. I just will kick out these instances off my Training/Validation data.

**Comment by pierluigiferrari** [comment]
Anytime!
